src/core/Akka.Remote/Endpoint.cs

```diff
@@ -411,147 +414,174 @@ protected override void PostRestart(Exception reason)
-                            // does not cancel existing timers (see the "else" case).
-                            RescheduleAutoResend();
-                        }
-                        else
-                        {
-                            ScheduleAutoResend();
-                        }
+                    ResendNacked();
+                }
+            });
+            Receive<AttemptSysMsgRedelivery>(sysmsg =>
+            {
+                if (UidConfirmed) ResendAll();
+            });
+            Receive<Terminated>(terminated =>
+            {
+                _currentHandle = null;
+                Context.Parent.Tell(new EndpointWriter.StoppedReading(Self));
+                if (_resendBuffer.NonAcked.Any() || _resendBuffer.Nacked.Any())
+                    Context.System.Scheduler.ScheduleTellOnce(_settings.SysResendTimeout, Self,
+                        new AttemptSysMsgRedelivery(), Self);
+                GoToIdle();
+            });
+            Receive<GotUid>(g =>
+            {
+                _bailoutAt = null;
+                Context.Parent.Tell(g);
+                //New system that has the same address as the old - need to start from fresh state
+                UidConfirmed = true;
+                if (Uid.HasValue && Uid.Value != g.Uid) Reset();
+                Uid = g.Uid;
+                ResendAll();
+            });
+            Receive<EndpointWriter.StopReading>(stopped =>
+            {
+                _writer.Forward(stopped); //forward the request
+            });
+        }
 
-                        ResendNacked();
-                    }
-                })
-                .With<AttemptSysMsgRedelivery>(sysmsg =>
-                {
-                    if (UidConfirmed) ResendAll();
-                })
-                .With<Terminated>(terminated =>
-                {
-                    _currentHandle = null;
-                    Context.Parent.Tell(new EndpointWriter.StoppedReading(Self));
-                    if (_resendBuffer.NonAcked.Count > 0 || _resendBuffer.Nacked.Count > 0)
-                        Context.System.Scheduler.ScheduleTellOnce(_settings.SysResendTimeout, Self, 
-                            new AttemptSysMsgRedelivery(), Self);
-                    Context.Become(Idle);
-                })
-                .With<GotUid>(g =>
-                {
-                    Context.Parent.Tell(g);
-                    //New system that has the same address as the old - need to start from fresh state
-                    UidConfirmed = true;
-                    if (Uid.HasValue && Uid.Value != g.Uid) Reset();
-                    else UnstashAcks();
-                    Uid = _refuseUid;
-                })
-                .With<EndpointWriter.StopReading>(stopped =>
-                {
-                    _writer.Forward(stopped); //forward the request
-                });
+        private void GoToIdle()
```

Comment:
Transition method called when the `ReliableDeliverySupervisor` needs to enter its `IdleBehavior`.

It will schedule a timer, whose duration is measured in **days**, that it's been unable to talk to the other side of the association, presumably because it or we are quarantined. 

Idle behavior is entered when:

1. The `EndpointWriter` is terminated during the `ReliableDeliverySupervisor`'s normal receiving behavior OR
2. The `ReliableDeliverySupervisor` is currently `Gated` (meaning that the association was marked as failed at some point), the `EndpointWriter` was never terminated, and the `nonAcked` or `nacked` buffers have NO messages in them.

I'll explain what the Idle behavior does below.